packages/web3-core/src/web3_request_manager.ts

```diff
@@ -219,42 +272,80 @@ export class Web3RequestManager<
 	private _processJsonRpcResponse<ResultType, ErrorType, RequestType>(
 		payload: JsonRpcPayload<RequestType>,
 		response: JsonRpcResponse<ResultType, ErrorType>,
+		{ legacy, error }: { legacy: boolean; error: boolean },
```

Comment:
can you create a type for `{ legacy: boolean; error: boolean }` and use that in this PR. It will be more clean.

Reply:
This type is created and used only here. So seems fine to just keep inline.